src/utils/gasPriceManager.ts

```diff
@@ -0,0 +1,450 @@
+        } catch (e) {
+            sentry.captureException(e)
+            this.logger.error(
+                "failed to fetch eip-1559 gasPrices from estimateFeesPerGas",
+                { error: e }
+            )
+            maxFeePerGas = undefined
+            maxPriorityFeePerGas = undefined
+        }
+
+        if (maxPriorityFeePerGas === undefined) {
+            this.logger.warn(
+                "maxPriorityFeePerGas is undefined, using fallback value"
+            )
+            try {
+                maxPriorityFeePerGas =
+                    await this.getFallBackMaxPriorityFeePerGas(
+                        this.publicClient,
+                        maxFeePerGas ?? 0n
+                    )
+            } catch (e) {
+                this.logger.error("failed to get fallback maxPriorityFeePerGas")
+                sentry.captureException(e)
+                throw e
+            }
+        }
+
+        if (maxFeePerGas === undefined) {
+            this.logger.warn("maxFeePerGas is undefined, using fallback value")
+            try {
+                maxFeePerGas =
+                    (await this.getNextBaseFee(this.publicClient)) +
+                    maxPriorityFeePerGas
+            } catch (e) {
+                this.logger.error("failed to get fallback maxFeePerGas")
+                sentry.captureException(e)
+                throw e
+            }
+        }
+
+        if (maxPriorityFeePerGas === 0n) {
+            maxPriorityFeePerGas = maxFeePerGas / 200n
+        }
+
+        return { maxFeePerGas, maxPriorityFeePerGas }
+    }
+
+    private saveMaxFeePerGas(gasPrice: bigint, timestamp: number) {
+        const queue = this.queueMaxFeePerGas
+        const last = queue.length > 0 ? queue[queue.length - 1] : null
+
+        if (!last || timestamp - last.timestamp >= 1000) {
+            if (queue.length >= this.maxQueueSize) {
+                queue.shift()
+            }
+            queue.push({ maxFeePerGas: gasPrice, timestamp })
+        } else if (gasPrice < last.maxFeePerGas) {
+            last.maxFeePerGas = gasPrice
+            last.timestamp = timestamp
+        }
+    }
+
+    private saveMaxPriorityFeePerGas(gasPrice: bigint, timestamp: number) {
+        const queue = this.queueMaxPriorityFeePerGas
+        const last = queue.length > 0 ? queue[queue.length - 1] : null
+
+        if (!last || timestamp - last.timestamp >= 1000) {
+            if (queue.length >= this.maxQueueSize) {
+                queue.shift()
+            }
+            queue.push({ maxPriorityFeePerGas: gasPrice, timestamp })
+        } else if (gasPrice < last.maxPriorityFeePerGas) {
+            last.maxPriorityFeePerGas = gasPrice
+            last.timestamp = timestamp
+        }
+    }
+
+    private saveGasPrice(gasPrice: GasPriceParameters, timestamp: number) {
+        return new Promise<void>((resolve) => {
+            this.saveMaxFeePerGas(gasPrice.maxFeePerGas, timestamp)
```

Comment:
why not save maxFeePerGas and maxPriorityFeePerGas together?

Reply:
because then comparing becomes difficult, you will have to choose one parameter to sort/save on the based of